### PR TITLE
Fix decoding hex values in unicode escapes

### PR DESCRIPTION
**Pull request by dsmatter**
Avoids decoding invalid strings like `"\ughij"`

### CODE REVIEW

File: parser/src/test/scala/jawn/SyntaxCheck.scala [hunk 1]
```diff
@@ -92,7 +92,15 @@ class SyntaxCheck extends Properties("SyntaxCheck") {
   }
 
   property("invalid unicode is invalid") = {
-    isValidSyntax("\"\\uqqqq\"") != true
+    isValidSyntax("\"\\uqqqq\"") != true &&
+    isValidSyntax("\"\\ughij\"") != true &&
```
Comment (travisbrown) [inline parser/src/test/scala/jawn/SyntaxCheck.scala]: Would you mind adding tests for `"\uGGGG"` and `"\ugggg"` here, just to verify that the boundary is exactly right for both lower and uppercase?

**Comment by non** [comment]
LGTM 👍

**Comment by travisbrown** [comment]
I'll backport this for an 0.14.3 bugfix release if there are no objections. /cc @larsrh

**Comment by travisbrown** [comment]
Here's the backport branch: https://github.com/typelevel/jawn/tree/backport/0.14.x
